### PR TITLE
Use a datetime format for parsing datetimes

### PR DESCRIPTION
**Pull request by christopher-dG**

### CODE REVIEW

File: src/read.jl [hunk 1]
```diff
@@ -205,7 +205,7 @@ function read(io::IO, T::Type{Char})
     return c
 end
 read(io::IO, ::Type{Dates.Date}, format=Dates.ISODateFormat) = Dates.Date(read(io, String), format)
-read(io::IO, ::Type{Dates.DateTime}, format=Dates.ISODateFormat) = Dates.DateTime(read(io, String), format)
+read(io::IO, ::Type{Dates.DateTime}, format=Dates.ISODateTimeFormat) = Dates.DateTime(read(io, String), format)
```
Comment (quinnj) [inline src/read.jl]: I'm not sure I understand the reasoning here? Don't we want to use the `DateTime` format for DateTimes?

Comment (christopher-dG) [reply src/read.jl]: You might be reading the diff backwards, this changes the format from `ISODateFormat` *to* `ISODateTimeFormat`. In the current implementation, both dates and datetimes are parsed with the same format string (which does not work for datetimes).

Comment (quinnj) [reply src/read.jl]: Doh! I thought I was missing something obvious. Ok, sounds good. Would you mind adding a test case that fails under current code and is fixed by the change here? You can just put it at the bottom of the `test/runtests.jl` file. It'd also be great if you could add a test for the failed `Union` parsing PR that I just merged.

Comment (christopher-dG) [reply src/read.jl]: Sure, will do this evening :slightly_smiling_face:.

**Comment by quinnj** [comment]
Thanks @christopher-dG! Sorry for being slow on these PRs.

**Pull request merged**